src/ui/components/map/providers/mapboxmapprovider.js

```diff
@@ -3,45 +3,71 @@
 import MapProvider from './mapprovider';
 import DOM from '../../../dom/dom';
 
-/* global mapboxgl */
+/* global mapboxgl, MapboxLanguage */
 
 /**
  * MapBoxMapProvider is an implementation of a MapProvider
  * that handles the integration with the third party API to expose maps.
  * @extends MapProvider
  */
 export default class MapBoxMapProvider extends MapProvider {
+  constructor (opts = {}, systemOpts = {}) {
+    super(opts, systemOpts);
+
+    this._isMapboxLoaded = false;
+    this._isMapboxLanguageLoaded = false;
+  }
   /**
    * Load the external JS Library
    * @param {function} onLoad An optional callback to invoke once the JS is loaded.
    */
   loadJS (onLoad) {
-    let script = DOM.createEl('script', {
+    let mapboxScript = DOM.createEl('script', {
       id: 'yext-map-js',
       onload: () => {
-        this._isLoaded = true;
-        mapboxgl.accessToken = this._apiKey;
-
-        if (typeof onLoad === 'function') {
-          onLoad();
-        }
-
-        if (typeof this._onLoaded === 'function') {
-          this._onLoaded();
-        }
+        this._isMapboxLoaded = true;
+        this._onScriptLoad(onLoad);
       },
       async: true,
       src: 'https://api.mapbox.com/mapbox-gl-js/v0.44.1/mapbox-gl.js'
     });
 
+    let mapboxLanguageScript = DOM.createEl('script', {
+      id: 'yext-map-language-js',
+      onload: () => {
+        this._isMapboxLanguageLoaded = true;
+        this._onScriptLoad(onLoad);
+      },
+      async: true,
+      src: 'https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-language/v0.10.1/mapbox-gl-language.js'
```

Comment:
it looks like this script is pretty small, maybe it would be easier and faster to just include it in the bundle? we have the upcoming plans to separate out the components so the hit to the bundle size will be pretty temporary

Reply:
Sure, that works